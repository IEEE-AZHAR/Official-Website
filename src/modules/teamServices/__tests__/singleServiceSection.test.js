import React from "react";

import { render, cleanup } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import "@testing-library/jest-dom/extend-expect";
import TeamServicesSection from "../components/teamServicesSection";

afterEach(() => cleanup());

describe("<TeamServicesSection /> tests", () => {
  it("should renders correctly", () => {
    let { container } = render(
      <Router>
        <TeamServicesSection />
      </Router>
    );

    expect(container).toMatchSnapshot();
  });

  it("should have 3 cards", () => {
    let { getByTestId } = render(
      <Router>
        <TeamServicesSection />
      </Router>
    );
    const servicesContainer = getByTestId("services-container");

    expect(servicesContainer.children.length).toEqual(3);
  });

  it("description should be 15 words only", () => {
    let { getAllByTestId } = render(
      <Router>
        <TeamServicesSection />
      </Router>
    );
    const serviceDescription = getAllByTestId("service-description")[0]
      .textContent;

    expect(serviceDescription.split(" ").length).toEqual(15);
  });

  it("'show more btn' should contains the correct link", () => {
    let { getAllByTestId } = render(
      <Router>
        <TeamServicesSection />
      </Router>
    );
    const serviceDescription = getAllByTestId("service-link")[0].href;

    expect(serviceDescription).toBe(window.location.origin + "/services/1");
  });
});
