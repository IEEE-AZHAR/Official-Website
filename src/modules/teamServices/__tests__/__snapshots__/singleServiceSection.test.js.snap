// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamServicesSection /> tests should renders correctly 1`] = `
<div>
  <section
    class="team-service-section"
  >
    <h2
      class="team-service-section_heading"
    >
      services
    </h2>
    <div
      class="container"
    >
      <div
        class="row"
        data-testid="services-container"
      >
        <div
          class="col-lg-4 col-md-6 col-sm-12 overflow-hidden text-center"
        >
          <a
            class="team-service-section_container"
            href="/services/1"
          >
            <img
              alt="Service Cover"
              class="team-service-section_cover"
              src="https://via.placeholder.com/500?text=Service+1"
            />
            <div
              class="team-service-section_content"
            >
              <h3
                class="team-service-section_title"
              >
                Service 1
              </h3>
              <p
                class="team-service-section_description"
              >
                <span
                  data-testid="service-description"
                >
                  Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptas, sed voluptate. Iste officia possimus necessitatibus
                </span>
                ...
              </p>
              <a
                class="team-service-section_button"
                data-testid="service-link"
                href="/services/1"
                tabindex="-1"
              >
                Read more about this service
              </a>
            </div>
          </a>
        </div>
        <div
          class="col-lg-4 col-md-6 col-sm-12 overflow-hidden text-center"
        >
          <a
            class="team-service-section_container"
            href="/services/2"
          >
            <img
              alt="Service Cover"
              class="team-service-section_cover"
              src="https://via.placeholder.com/500?text=Service+2"
            />
            <div
              class="team-service-section_content"
            >
              <h3
                class="team-service-section_title"
              >
                Service 2
              </h3>
              <p
                class="team-service-section_description"
              >
                <span
                  data-testid="service-description"
                >
                  Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptas, sed voluptate. Iste officia possimus necessitatibus
                </span>
                ...
              </p>
              <a
                class="team-service-section_button"
                data-testid="service-link"
                href="/services/2"
                tabindex="-1"
              >
                Read more about this service
              </a>
            </div>
          </a>
        </div>
        <div
          class="col-lg-4 col-md-6 col-sm-12 overflow-hidden text-center"
        >
          <a
            class="team-service-section_container"
            href="/services/3"
          >
            <img
              alt="Service Cover"
              class="team-service-section_cover"
              src="https://via.placeholder.com/500?text=Service+3"
            />
            <div
              class="team-service-section_content"
            >
              <h3
                class="team-service-section_title"
              >
                Service 3
              </h3>
              <p
                class="team-service-section_description"
              >
                <span
                  data-testid="service-description"
                >
                  Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptas, sed voluptate. Iste officia possimus necessitatibus
                </span>
                ...
              </p>
              <a
                class="team-service-section_button"
                data-testid="service-link"
                href="/services/3"
                tabindex="-1"
              >
                Read more about this service
              </a>
            </div>
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
